File: Python/single_threaded.py
import cv2
import numpy as np
from typing import Final

from imageproc_utils import recolor_image


OUTPUT_IMAGE: Final[str] = 'result_image/single-threaded.jpg'


def single_threaded(img: np.ndarray) -> None:
    Width: Final[int] = img.shape[1]
    Height: Final[int] = img.shape[0]
    recolor_image(img, img, 0, 0, Width, Height)

    cv2.imwrite(OUTPUT_IMAGE, img)

File: C++/src/single_threaded.cpp
#include "single_threaded.hpp"
#include "image_processing.hpp"
#include <opencv2/imgcodecs.hpp>

namespace single_threaded
{

  void processImage(cv::Mat &img)
  {
    color_changer::recolorImage(img, img, 0, 0, img.cols, img.rows);
    cv::imwrite("../../result_image/single-threaded.jpg", img);
  }

} // namespace single_threaded
